[Bazel/MSVC] Cleanup flags and config_settings for MSVC

rongjiecomputer opened the pull request.

rongjiecomputer committed:
[Bazel/MSVC] Cleanup -ldl/-lm flags for MSVC

rongjiecomputer committed:
More cleanup

rongjiecomputer commented:
PTAL @mrry

meteorcloudy commented:
@rongjiecomputer Thank you so much for doing those clean up work!
Could you also remove all usage of `//tensorflow:windows_msvc` config_settting? Because that's a legacy config_setting which is now identical to `//tensorflow:windows`.

meteorcloudy commented on third_party/snappy.BUILD:
```diff
@@ -18,17 +18,10 @@ cc_library(
-        ],
+    copts = ["-DHAVE_CONFIG_H"] + select({
+        "@org_tensorflow//tensorflow:windows": [],
+        "@org_tensorflow//tensorflow:windows_msvc": [],
```
We can remove this

rongjiecomputer committed:
Remove all references of windows_msvc config_setting

rongjiecomputer commented:
> Could you also remove all usage of //tensorflow:windows_msvc config_settting? Because that's a legacy config_setting which is now identical to //tensorflow:windows.

Done. I have removed all references to `:windows_msvc` I can find with Github search. Let's see if this break CI build.

rongjiecomputer committed:
Remove windows_msvc from png_internal

rongjiecomputer commented:
@meteorcloudy After our last conversation, one more `windows_msvc` slipped into the codebase https://github.com/tensorflow/tensorflow/commit/158cd6220231fcf758a45c2dcd40d93cd0aec9e0#diff-15609f78c0cd35bfdc31f9d8bd9b1e91R2171.

I have rebased and removed the `windows_msvc`.

Please help to spread awareness to internal team to not use `windows_msvc` anymore.

martinwicke commented:
@meteorcloudy maybe we can use a METADATA warning to prevent new occurrences.

rongjiecomputer committed:
Fix buildifier check

meteorcloudy commented:
Thanks @martinwicke , will do it!

tensorflowbutler commented:
Nagging Assignee @martinwicke: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

rongjiecomputer committed:
Merge branch 'master' into flag

rongjiecomputer commented:
Looks like my change is pushed from internal already. Closing this now. 😄
